src/ContextualPopup/ContextualPopup.js

```diff
@@ -222,6 +229,18 @@ module.exports = kind(
 	initComponents: function () {
 		this.createChrome(this.tools);
 		Popup.prototype.initComponents.apply(this, arguments);
+		this.syncProperties();
+	},
+
+	/**
+	* Sync spotlightModal and modal properties on creation time.
+	*
+	* @private
+	*/
+	syncProperties: function () {
+		if (this.modal == this.spotlightModal) return;
+		if (this.hasOwnProperty('spotlightModal')) this.modal = this.spotlightModal;
```

Comment:
Thanks for the clarification! I don't think we can make `spotlightModal` private, as that affects the enabling of the close button, and developers would still need a mechanism for controlling the close button. I think for that reason, and because `spotlightModal` is the more "specific" property (`modal` is more general and is inherited from `enyo/Popup`), it seems to make the most sense to give `spotlightModal` priority in terms of sync direction (i.e. sync from `spotlightModal` to `modal`).

It wasn't clear to me that we needed or wanted to support "overriding" of one or both of the properties at create time, as the original issue could then reoccur. Are there specific cases you know of where we need to support this? If not, we can simplify the syncing logic to `this.modal = this.spotlightModal`. I think we would want to be as consistent as possible in our handling of these properties, if we are going to make a change.